container/build.py

```diff
@@ -99,32 +105,66 @@
         "bazel_{}_docker-packages.tar".format(LATEST_BAZEL_VERSION),
 }
 
-assert set(supported_types) \
+assert set(SUPPORTED_TYPES) \
         == set(TYPE_PACKAGE_MAP.keys()) \
         == set(TYPE_TARGET_MAP.keys()) \
         == set(TYPE_TARBALL_MAP.keys()), \
             "TYPES ARE OUT OF SYNC"
 
 
-def main(type_, project, container, tag, async_, bucket, local):
+def main(type_, project, container, tag, async_, bucket, local, bazel_version, map = None):
   '''Runs the build. More info in module docstring at the top.
   '''
+  
+  if map: # Override the map values
+    map_module = imp.load_source("map", map)
+    TYPE_PACKAGE_MAP = map_module.TYPE_PACKAGE_MAP
+    TYPE_TARGET_MAP = map_module.TYPE_TARGET_MAP
+    TYPE_TARBALL_MAP = map_module.TYPE_TARBALL_MAP
+
+  # Gets the project root (for calling bazel targets)
   project_root = subprocess.check_output(
       shlex.split("git rev-parse --show-toplevel")).strip()
   package = TYPE_PACKAGE_MAP[type_]
   target = TYPE_TARGET_MAP[type_]
-  tarball = TYPE_TARBALL_MAP[type_]
+  tarball = None
+  if bucket:
+    tarball = TYPE_TARBALL_MAP[type_]
+
+  # Gets the base directory of the bazel-toolchains repo relative to this 
+  # build.py. This is for referencing yaml files and mounting project to gcloud
+  # THIS NEEDS TO BE UPDATED IF THIS FILE IS MOVED 
+  bazel_toolchains_base_dir = os.path.relpath(os.path.join(__file__, "../.."))
 
   # We need to start the build from the root of the project, so that we can
   # mount the full root directory (to use bazel builder properly).
   os.chdir(project_root)
   # We need to run clean to make sure we don't mount local build outputs
   subprocess.call(["bazel", "clean"])
 
+  # Ensure all BUILD files under /third_party have the right permission.
+  # This is because in some systems the BUILD files under /third_party
+  # (after git clone) will be with permission 640 and the build will
+  # fail in Container Builder.
+  for dirpath, _, files in os.walk("third_party"):
+    for f in files:
+      full_path = os.path.join(dirpath, f)
+      os.chmod(full_path, 0o644)
+
   if local:
     local_build(type_, package, target)
   else:
-    cloud_build(project_root, project, container, tag, async_, bucket, package, target, tarball)
+    # Gets the yaml relative to this build.py, regardless of what directory it was called from
```

Comment:
could we receive this info as an optional argument (or somehow make it easier to maintain it so that the detail that the paths must be kept in sync is more visibly)

Reply:
Changed it to find the base of bazel-toolchains using the "git rev-parse --show-toplevel" command, so the location of the build.py no longer matters.